var/spack/repos/builtin/packages/py-celery/package.py

```diff
@@ -46,19 +47,26 @@ class PyCelery(PythonPackage):
     # 'cosmosdbsql',
     # 's3',
 
+    depends_on('python@3.7:', type=('build', 'run'), when="@5.2.3:")
     depends_on('python@2.7:2.8,3.4:', type=('build', 'run'))
 
     depends_on('py-setuptools', type=('build', 'run'))
 
     depends_on('py-redis@3.2.0:', when='+redis', type=('build', 'run'))
     depends_on('py-sqlalchemy', when='+sqlalchemy', type=('build', 'run'))
 
-    depends_on('py-click@7.0:7', when="@5.0.0:5", type=('build', 'run'))
+    depends_on('py-click@7.0:7', when="@5.0.0:5.0", type=('build', 'run'))
+    depends_on('py-click@8.0.3:8', when="@5.2.0:5.2", type=('build', 'run'))
```

Comment:
You can probably leave off the upper bound on these `when='...'` args since newer versions may require the same versions of these deps

Reply:
Can do, I'm concerned that celery 5.0 may not work right with click 8.x.  I've not tested that at all.